test/inductor/test_compiled_optimizers.py

```diff
@@ -84,12 +84,12 @@ class KernelCounts(NamedTuple):
 KERNEL_COUNTS = {
     Adam: KernelCounts(multitensor=2, singletensor=8),
     AdamW: KernelCounts(multitensor=2, singletensor=8),
-    NAdam: KernelCounts(multitensor=2, singletensor=12),
+    NAdam: KernelCounts(multitensor=2, singletensor=8),
```

Comment:
Tracing optim.state first instead of tracing optim.param_groups changes input ordering for the AOT graphs. This affects the input graph for inductor, which leads to different fusions.

Reply:
Yeah this is great! I was wondering why NAdam was so bad tbh